docs/userguide/release-version.md

```diff
@@ -0,0 +1,17 @@
+# Opensearch Operator New version release guide
```

Comment:
I think this entire file should not be part of the userguide (as it is not something for users). My preference would be to dump the contents into the MAINTAINERS.md file